src/snippets/menu.liquid

```diff
@@ -0,0 +1,158 @@
+{%- comment -%}
+
+  This snippet renders a menu with support for dropdowns
+
+  Parameters:
+  - menu: the navigation menu object to render
+  - menu_back_label: label for the 'Back' button in mobile dropdowns (default is 'link.title')
+  - menu_levels: number of menu nested dropdown menus to display (max. 3 levels)
+  - section_class: a CSS class prefix for styling the menu
+  - settings: theme settings
+  - socials: social links to include at the bottom of the mobile menu (default is true)
+  - use_opener: whether to include a menu opener on mobile, e.g. hamburger button to open menu (default is true)
+
+  Required parameters:
+  - menu
+  - section_class
+  - settings
+
+  Usage:
+
+  {%- render 'menu',
+      menu: main_menu,
+      menu_back_label: 'Go back',
+      menu_levels: number,
+      section_class: 'your-class',
+      settings: settings,
+      socials: false,
+      use_opener: false
+  -%}
+{%- endcomment -%}
+
+{%- assign menu = menu | default: '' -%}
+
+{%- if menu != blank -%}
+  {%- assign class = 'menu' -%}
+  {%- assign content = content | default: '' -%}
+  {%- assign menu_levels = menu_levels | default: 1 -%}
+  {%- assign menu_items = menu.items -%}
+  {%- assign socials = socials | default: true -%}
+  {%- assign use_opener = use_opener | default: true -%}
+
+  {%- capture icon_arrow -%}
+    <i class="menu__item-icon">
+      {%- render 'icon-arrow' -%}
+    </i>
+  {%- endcapture -%}
+
+  <div id="{{- section_class -}}-{{- class -}}" class="{{- section_class -}}__{{ class }} {{ class }}">
```

Comment:
Do we plan to use this menu anywhere outside of the header?
I'm not sure that this approach is worth it, it makes code very hard to read. Can't we just have normal hard-coded classes everywhere?
If I wanted to change the styling of this DIV i have no idea where to start as the CSS classes could be anything.